File: tests/fixtures/edgecase_repo/ruby/module5.rb
require_relative "./module4"

module Example
  class Base5
    def initialize(name)
      @name = name
    end
  end

  class Service5 < Base5
    def handle(payload)
      validate(payload)
      true
    end

    def validate(payload)
      !payload.nil?
    end
  end
end

def helper_5(value)
  value
end

File: tests/fixtures/edgecase_repo/php/module5.php
<?php
namespace Example\Gen;

use Example\Gen\Base5;
use Example\Gen\Handler5 as H5;

class Base5
{
    protected $name;
}

interface Handler5
{
    public function handle($payload);
}

class Service5 extends Base5 implements Handler5
{
    public function handle($payload): bool
    {
        $this->validate($payload);
        return true;
    }

    private function validate($payload): void
    {
    }
}

function helper_5($value) {
    return $value;
}
